File: back-spirng/jobseeckerstudio/quiz/src/main/java/com.jobseeckerstudio.quiz/entity/category/Category.java
package com.jobseeckerstudio.user.quiz.entity.category;

import com.jobseeckerstudio.user.quiz.entity.common.CategoryName;
import com.jobseeckerstudio.user.quiz.entity.quiz.Quiz;

import javax.persistence.*;
import java.util.List;

@Entity
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "category_id")
    private int categoryId;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private CategoryName categoryName;

    @OneToMany(mappedBy = "category")
    private List<Quiz> quizzes;
}

File: back-node/src/domain/category/Category.ts
import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';

@Entity()
export class Category {
  @PrimaryGeneratedColumn()
  private readonly categoryId: number | undefined;

  @Column()
  private readonly categoryName: string;

  constructor(categoryName: string, categoryId?: number) {
    this.categoryId = categoryId;
    this.categoryName = categoryName;
  }
}
